### PR TITLE
Newmetric: Jensen-Shannon Divergence

### LINKED ISSUE #2947: Adding Jensen-Shannon (JS) Divergence Metric
## 🚀 Feature

Given two probability distributions $P$ and $Q$, the JS divergence is defined as:

$$JS(P \mid \mid Q) = \frac{1}{2} KL(P \mid \mid M) + \frac{1}{2} KL(Q \mid \mid M)$$

where:

$KL(P \mid \mid Q)$ is the Kullback-Leibler divergence:
  
  $$
  KL(P \mid \mid Q) = \sum_i P(i) . \log \frac{P(i)}{Q(i)}
  $$

$M$ is the mixture distribution (mean?), defined as

$M = \frac{1}{2} (P + Q)$

Thus, the JS divergence computes the KL divergence between each distribution and the average of both, making it **symmetric** and **bounded** in the range:

$0 \leq JS(P \mid \mid Q) \leq \log 2$

### Motivation

The PyTorch Metrics library currently includes Kullback-Leibler (KL) Divergence, which is widely used for measuring differences between probability distributions. However, KL divergence is asymmetric and unbounded, making it sometimes less suitable for certain applications.

Jensen-Shannon (JS) Divergence provides a symmetric and smoothed alternative by averaging KL divergence between two distributions and a mixed distribution. This makes it useful for various tasks, including generative modeling, NLP, and probabilistic machine learning.

Adding JS divergence to PyTorchmetrics would align with existing divergence metrics.

### Pitch

I propose adding JS Divergence as a new metric in PyTorch Metrics, similar to how KL Divergence is implemented. My implementation is already available in my forked repository, and I’d be happy to refine it based on community feedback.

### Alternatives

- Manual computation: Users can manually compute JS divergence using KL divergence, but having a built-in makes it scalable.
- Third-party libraries: Some users rely on scipy.spatial.distance.jensenshannon. which I guess is the sqrt of the JSD.

### Additional context

N/A

### PR DESCRIPTION
**Pull request by SkafteNicki**
## What does this PR do?

Fixes #2947

<details>
  <summary>Before submitting</summary>

- [ ] Was this **discussed/agreed** via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/torchmetrics/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to **update the docs**?
- [ ] Did you write any new **necessary tests**?

</details>

<details>
  <summary>PR review</summary>

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

</details>

## Did you have fun?

Make sure you had fun coding 🙃

----
📚 Documentation preview 📚: https://torchmetrics--2992.org.readthedocs.build/en/2992/

### CODE REVIEW

File: src/torchmetrics/functional/regression/js_divergence.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+import torch
+from torch import Tensor
+from typing_extensions import Literal
+
+from torchmetrics.functional.regression.kl_divergence import kl_divergence
+from torchmetrics.utilities.checks import _check_same_shape
+
+
+def _jsd_update(p: Tensor, q: Tensor, log_prob: bool) -> tuple[Tensor, int]:
+    """Update and returns jensen-shannon divergence scores for each observation and the total number of observations.
```
Comment (Borda) [inline src/torchmetrics/functional/regression/js_divergence.py:26]: let's add a reference to the wiki?

**Review by Borda — APPROVED** [review]
